### PR TITLE
Endpoint for deleting bookings 139

### PR DESCRIPTION
**Pull request by jeffplays2005**
Generated routes & schemas.

Added request/response interfaces.

Created endpoint.

### CODE REVIEW

File: server/src/service-layer/controllers/AdminController.ts [hunk 1]
```diff
@@ -121,7 +128,7 @@ export class AdminController extends Controller {
         return { bookingSlotId: bookingSlotForDate.id, date: dateTimestamp }
       } catch (e) {
         console.error(
-          `Something went wrong when trying to make the date
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:124]: What's happening here

File: server/src/service-layer/controllers/AdminController.ts [hunk 2]
```diff
@@ -141,6 +148,25 @@ export class AdminController extends Controller {
...
+  public async removeBooking(
+    @Body() requestBody: DeleteBookingRequest
+  ): Promise<BookingDeleteResponse> {
+    const { bookingId } = requestBody
+    // Validate and check if the booking actually exists
+    const bookingDataService = new BookingDataService()
+    const { user_id } = await bookingDataService.getBookingById(bookingId)
+    if (!user_id) {
+      this.setStatus(404)
+      return { message: "Booking not found with that booking ID." }
```
Comment (bcho892) [inline server/src/service-layer/controllers/AdminController.ts:163]: Nit: early return means you don't need else

File: server/src/service-layer/request-models/AdminRequests.ts [hunk 3]
```diff
@@ -2,12 +2,12 @@ import { Timestamp } from "firebase-admin/firestore"
 
 export interface MakeDatesAvailableRequestBody {
   /**
-   * Firestore timestamp, should represent a UTC date that is set to exactly midnight
+   * Firestore timestamp, ideally with the time information removed (set to midnight)
    */
   startDate: Timestamp
 
   /**
-   * Firestore timestamp, should represent a UTC date that is set to exactly midnight
```
Comment (bcho892) [inline server/src/service-layer/request-models/AdminRequests.ts:10]: Dont think these changes are part of PR?

**Review by bcho892 — APPROVED** [review]

File: server/src/service-layer/request-models/AdminRequests.ts (hunk 3 above)
Comment (jeffplays2005) [reply server/src/service-layer/request-models/AdminRequests.ts:10]: Oh oops wtf my git is being weird again

File: server/src/service-layer/controllers/AdminController.ts (hunk 2 above)
Comment (jeffplays2005) [reply server/src/service-layer/controllers/AdminController.ts:165]: Ah, yep also noticed that and fixed in the later push

File: server/src/service-layer/request-models/AdminRequests.ts (hunk 3 above)
Comment (jeffplays2005) [reply server/src/service-layer/request-models/AdminRequests.ts]: Just made a push to address the weird changes

**Review by bcho892 — APPROVED** [review]

**Pull request merged**